Automattic/wp-calypso
# Memberships: Use number type for currency input

Related to 209-gh-automattic/gold

## Proposed Changes

* Allow a `type` prop to be sent to `FormTextInput` component.
* Use this `type` prop as the `input` `type`, defaulting to `"text"`
* Pass `type="number"` allow the browser to handle numerical symbols for `FormCurrencyInput` 

## Testing Instructions



To recreate the issue (without this PR applied):
* Visit https://wordpress.com/earn/payments-plans/ and select a site with payment plans or with Stripe connected.
* Edit an existing plan or create a new plan.
* Try entering numerical symbols such as `.` into the "Monthly Price" and "Annual Price" boxes. For example: If you want to set the value to `20.50`, you can enter `20` but as soon as you type the `.` you'll see `20` in the field. The only way to enter a value of `20.50` is to enter `2050` then click the cursor between the `20` and the `50` and place the decimal there. This is because the value is passed through `parseFloat` with each text change, which removes the `.` in cases such as `20.`.

To fix the issue (with this PR applied):
* Use the calypso live links below and navigate to `/earn/payments-plans` (Tools > Monetize). Select a site with payment plans, or with Stripe connected.
* Edit an existing plan or create a new plan.
* Try entering `20.50` or some other number with a decimal. This should work as expected.

## Pre-merge Checklist
